### PR TITLE
Fix calls to release notes from cloud instances

### PR DESCRIPTION
**Pull request by sharat87**

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java [hunk 1]
```diff
@@ -40,6 +40,12 @@
     @Value("${github_repo}")
     private String repo;
 
+    @Value("${is.cloud-hosted:false}")
+    private boolean isCloudHosted;
+
+    @Value("${segment.ce.key}")
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java:46]: Is it possible to read these values from a common config class? I'm generally wary of parsing the same configuration value from the properties file in multiple locations in the source code. This leads to errors in default values, spelling mistakes etc.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java [hunk 2]
```diff
@@ -76,6 +82,7 @@ public ReleaseNode(String tagName) {
                 .flatMap(instanceId -> WebClient
                         .create(
                                 baseUrl + "/api/v1/releases?instanceId=" + instanceId +
+                                        "&isSourceInstall=" + (isCloudHosted || StringUtils.isEmpty(segmentKey)) +
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java:85]: Should this be `!isCloudHosted`?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java:85]: Cloud hosted instances should behave *equal* to source-installed instances right? Both shouldn't send the fetch-release-notes data point.

Comment (mohanarpit) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java:85]: Oh ya. Good point. :)

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java (hunk 1 above)
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ReleaseNotesService.java]: I moved the segment key to a common place, but the is-cloud-hosted one is being used in a different way in the other place, I don't think it makes sense to do a refactor there for this. Does that sound okay?

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**